sdk/src/resource/resource_detector.cc

```diff
@@ -15,28 +15,42 @@ namespace resource
 {
 
 const char *OTEL_RESOURCE_ATTRIBUTES = "OTEL_RESOURCE_ATTRIBUTES";
+const char *OTEL_SERVICE_NAME        = "OTEL_SERVICE_NAME";
 
 Resource OTELResourceDetector::Detect() noexcept
 {
-  std::string attributes_str;
-  bool exists;
+  std::string attributes_str, service_name;
 
-  exists = opentelemetry::sdk::common::GetStringEnvironmentVariable(OTEL_RESOURCE_ATTRIBUTES,
-                                                                    attributes_str);
-  if (!exists)
+  bool attributes_exists = opentelemetry::sdk::common::GetStringEnvironmentVariable(
+      OTEL_RESOURCE_ATTRIBUTES, attributes_str);
+  bool service_name_exists =
+      opentelemetry::sdk::common::GetStringEnvironmentVariable(OTEL_SERVICE_NAME, service_name);
+
+  if (!attributes_exists && !service_name_exists)
   {
     return Resource();
   }
+
   ResourceAttributes attributes;
-  std::istringstream iss(attributes_str);
-  std::string token;
-  while (std::getline(iss, token, ','))
+
+  if (attributes_exists)
   {
-    size_t pos        = token.find('=');
-    std::string key   = token.substr(0, pos);
-    std::string value = token.substr(pos + 1);
-    attributes[key]   = value;
+    std::istringstream iss(attributes_str);
+    std::string token;
+    while (std::getline(iss, token, ','))
+    {
+      size_t pos        = token.find('=');
+      std::string key   = token.substr(0, pos);
+      std::string value = token.substr(pos + 1);
+      attributes[key]   = value;
+    }
```

Comment:
This is unrelated to your fix. If you can replace the above while loop as below (just adding extra validation):
```cpp
    while (std::getline(iss, token, ','))
    {
      size_t pos = token.find('=');
      if (pos != std::string::npos) {
        std::string key = token.substr(0, pos);
        std::string value = token.substr(pos + 1);
        attributes[key] = value;
      }
    }
```
If not, I will fix it separately :)

Reply:
Added to latest commit.